### PR TITLE
Build mhld data

### PR DESCRIPTION
**Pull request by jcoyne**

### CODE REVIEW

File: lib/folio_record.rb [hunk 1]
```diff
@@ -91,6 +92,35 @@ def self.folio_sirsi_locations_map
...
+
+  def mhld
+    items_per_holding.map do |holding|
+      library, location = self.class.folio_sirsi_locations_map.fetch(holding.fetch(:location).fetch(:code))
+      public_note = holding.fetch(:note)
+      library_has = holding.fetch(:library_has)
+      latest_received = holding.fetch(:items).map do |item|
+        # OMG this makes so many requests!!!
+        item_id = item['id']
+        response = client.get_json("/orders/pieces?query=itemId==\"#{item_id}\"")
```
Comment (cbeer) [inline lib/folio_record.rb:116]: It looks like you might be able to pivot through titles?
e.g
> client.get_json("/orders/pieces?query=titles.instanceId==\"b0a12f64-5116-56a2-a0aa-0200b8745c26\"")

File: lib/folio_record.rb [hunk 2]
```diff
@@ -2,8 +2,10 @@
 
 require 'active_support/core_ext/module/delegation'
 require_relative 'traject/common/constants'
+require 'csv'
 
 class FolioRecord
+  LIMIT = (2**31) - 1 # Folio max results
```
Comment (thatbudakguy) [inline lib/folio_record.rb:8]: this is also a magic number here: https://github.com/sul-dlss/searchworks_traject_indexer/blob/master/lib/traject/readers/folio_reader.rb#L19

File: lib/folio_record.rb [hunk 3]
```diff
@@ -91,6 +93,50 @@ def self.folio_sirsi_locations_map
...
+  end
+
+  # Look at the journal Nature (hrid: a3195844) as a pathological case (but pieces aren't loaded there yet)
+  # hrid: a567006 has > 1000 on test.
+  def pieces_per_holding
+    @pieces_per_holding ||= record.fetch('pieces') { pieces_from_api }.compact.group_by { |piece| piece['holdingId'] }
+  end
+
+  def pieces_from_api
+    client.get_json("/orders/pieces?limit=#{LIMIT}&query=titles.instanceId==\"#{instance_id}\"")
```
Comment (cbeer) [inline lib/folio_record.rb:136]:
```suggestion
    client.get_json("/orders/pieces", params: { limit: LIMIT, query: "titles.instanceId==\"#{instance_id}\""})
```

**Review by thatbudakguy — APPROVED** [review]

File: lib/folio_record.rb [hunk 4]
```diff
@@ -2,8 +2,10 @@
 
 require 'active_support/core_ext/module/delegation'
 require_relative 'traject/common/constants'
+require 'csv'
```
Comment (cbeer) [inline lib/folio_record.rb:5]:
```suggestion
```

I'm not sure where CSV is used?

**Pull request merged**